core/graph/src/main/java/uk/gov/gchq/gaffer/graph/hook/AddOperationsToChain.java

```diff
@@ -109,25 +114,25 @@ public void setAuthorisedOps(final LinkedHashMap<String, AdditionalOperations> a
         }
     }
 
-    public LinkedHashMap<String, AdditionalOperations> getAuthorisedOps() {
-        return authorisedOps;
-    }
-
-    private List<Operation> addOperationsToChain(final OperationChain<?> opChain, final AdditionalOperations additionalOperations) {
+    private List<Operation> addOperationsToChain(final OperationChain<?> operation, final AdditionalOperations additionalOperations) {
         List<Operation> newOpList = new ArrayList<>();
 
         newOpList.addAll(additionalOperations.getStart());
-        if (opChain != null && !opChain.getOperations().isEmpty()) {
-            for (final Operation originalOp : opChain.getOperations()) {
-                if(originalOp instanceof OperationChain) {
+        if (operation != null && !operation.getOperations().isEmpty()) {
```

Comment:
Please could you add some tests to check nested operations are handled correctly.